food-truck-frontend/src/api/Truck.ts

```diff
@@ -48,3 +48,12 @@ export const getTruckById = async (truckId: number, onFail: (err: any) => void)
     }
     return undefined;
 }
+
+export const searchTruckByName = async (search: string, onFail: (err: any) => void) => {
+  let result = await api.get(`/truck/search?search=${search}`)
```

Comment:
I'll admit I'm still not 100% sure how mixing async/await and then/catch work in Javascript. If it let us do this to handle errors and stuff in 1 line, that'd be awesome. If it seems to be working now, we can just assume it does work and not worry about it.